lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -280,11 +332,103 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         this.categoryControlVisible = false;
     }
 
-    cancelChanges() {
+    cancelChanges(buttonType: ButtonType, event: MouseEvent) {
+        event.stopPropagation();
+        this.toggleEditMode(buttonType);
         this.revertChanges();
         this.loadProperties(this.node);
     }
 
+    isEditingPanel(): boolean {
+        return (
+            (this.editable && this.hasMetadataChanged) ||
+            (this.editableTags  && this.hasMetadataChanged ) ||
+            (this.editableCategories  && this.hasMetadataChanged) ||
+            ((this.editableGroup?.editable)  && this.hasMetadataChanged)
+        );
+    }
+
+    showSnackbar(message: string): void {
+        this.snackBar.open(message, '', {
+            duration: 3000,
+            verticalPosition: 'bottom',
+            panelClass: ['adf-snackbar-message']
+        });
+    }
+
+    toggleEdit(event: MouseEvent, group: CardViewGroup, buttonType: ButtonType): void {
+        if (this.isEditingPanel()) {
+            this.showSnackbar('METADATA.BASIC.SNACKBAR_MESSAGE');
+            return;
+        }
+
+        if (this.editableGroup && this.editableGroup !== group) {
+            this.editableGroup.editable = false;
+        }
+
+        event.stopPropagation();
+    
+        switch (buttonType) {
+            case ButtonType.GeneralInfo:
+                this.editable = !this.editable;
+                this.panel.open();
+                this.editableTags = false;
+                this.editableCategories = false;
+                break;
+    
+            case ButtonType.Tags:
+                this.editableTags = !this.editableTags;
+                this.tagsPanelState = this.editableTags;
+                this.tagNameControlVisible = true;
+                break;
+    
+            case ButtonType.Categories:
+                this.editableCategories = !this.editableCategories;
+                this.categoriesPanelState = this.editableCategories;
+                this.categoryControlVisible = true;
+                break;
+    
+            case ButtonType.Group:
+                group.editable = !group.editable;
+                this.editableChange.emit(this.editable);
+                this.editableGroup = group.editable ? group : null;
+                if (group.editable) {
+                    group.expanded = true;
+                }
+                break;
+    
+            default:
+                break;
+        }
+    
+        if (buttonType !== ButtonType.GeneralInfo) {
```

Comment:
Please fix eslint trailing spaces issues in that file: 
<img width="672" alt="image" src="https://github.com/Alfresco/alfresco-ng2-components/assets/115619721/6374d83b-a7f4-4625-9423-045bbc58967b">

Reply:
I have fixed linting issues.